packages/application-generic/src/services/auth/auth.guard.ts

```diff
@@ -26,6 +26,11 @@ export class JwtAuthGuard extends AuthGuard('jwt') {
     const request = context.switchToHttp().getRequest();
     const authorizationHeader = request.headers.authorization;
 
+    if (authorizationHeader) {
+      const authScheme = authorizationHeader.split(' ')[0];
+      request.authScheme = authScheme;
+    }
+
```

Comment:
This is a little ugly right now. It's required because we currently override the existing auth header when an API key is used to authenticate. Once we remove this override, we can let the downstream execution flow read the auth header again as normal.

It will be fixed in https://linear.app/novu/issue/NV-3055/🔐-fix-apikey-auth-guard-performance

Reply:
I will make sure to prioritize NV-3055 for the next cycle